lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1051,6 +1052,15 @@ bool LocalEnforcer::update_tunnel_ids(
     session->receive_monitor(monitor, nullptr);
   }
 
+  // iterate through sessions in session map and update shard ids
+  SessionMap::iterator iter;
+  for (iter = session_map.begin(); iter != session_map.end(); iter++) {
+    int shard_id = shards_->add_ue();
```

Comment:
This seems like it will just reassign every ue a new shard, but a UE should already have a shard associated with it. Can we reconstruct state here?

Reply:
It'll assign every UE _session_ a new shard. The reason I do it here is because there is no object definition of a UE and I want to avoid giving each session it's own shard. It would be ideal for a UE to already have a shard associated with it, but I didn't see the place to make the addition(Oriol suggested the update_tunnel_ids function, which is why I've put it in here)

Reply:
Don't we save it in session state
marshaled["shard_id"]          = stored.shard_id;

Reply:
We need to recover what we previously assigned its a hard requirement, or pipelined state will get recked

Reply:
Yeah we do, but it's never assigned. Assignment happens when we add a UE. I thought if I assigned it there that it would add a "new UE" per session instead of per UE, which is why I didn't do marshaled["shard_id"] = add_ue();

Reply:
This looks wrong. `session_map` contains all existing sessions in the system.  Why do we have to change shard id 

Plus if the idea is to maintain this in both sessiond and pipelined, that means that if we have 1000 sessions, we will have to update 1000 ues on sessiond and then send an update to pipelined with to inform of the new shard id. 

Update Tunnel applies to one single session (defined at `auto& session = **session_it;`). But this functions seems to iterate over the whole session map.

Reply:
You are right. The idea was to set shard id after session creation because of the issue of where to put the shard object and avoiding creating duplicate entries for multiple sessions in a single UE. I thought this would be a good place to set the shard id. I'm working on a new approach.

Reply:
I'll be handling state recovery in a separate PR, but it should be easy to add in with this approach.